### ISSUE TITLE
Rserve 1.8-6 source installation with failed c complier

### ISSUE THREAD

**Issue by Yukiwang77** [s-u/Rserve#151]
Hi all,

I have been encountering this issue for the past two days and my BAU work is relying on a software that needs Rserve 1.8-6 for R integration. The issue started after I updated my R version to 4.0.1 and I am on mac os Catalina 10.15.4. The command I used for installation is 

> install.packages('Rserve',,"http://rforge.net/",type="source")

and the error pops up as

> * installing *source* package ‘Rserve’ ...
> ** using staged installation
> checking whether to compile the server... yes
> checking whether to compile the client... no
> checking for gcc... /usr/local/Cellar/gcc/8.3.0/bin/gcc-8
> checking whether the C compiler works... no
> configure: error: in `/private/var/folders/tm/gsxql1_j2zz23kt4c858p2mj_jz7py/T/RtmpV7KOOa/R.INSTALL3b35751cac82/Rserve':
> configure: error: C compiler cannot create executables
> See `config.log' for more details
> ERROR: configuration failed for package ‘Rserve’

For versions:
gcc --version

> Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
> Apple clang version 11.0.3 (clang-1103.0.32.62)
> Target: x86_64-apple-darwin19.4.0
> Thread model: posix
> InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
> 

clang --version

> Apple clang version 11.0.3 (clang-1103.0.32.62)
> Target: x86_64-apple-darwin19.4.0
> Thread model: posix
> InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

**Comment by Yukiwang77** [comment]
sorry but I dont think I got the solution quite yet. If the homebrew gcc is broken how do I fix it or how do i change the R configuration to use the right one?

**Comment by s-u** [comment]
That you have to find out - we don't know your setup is, as I said, `config.log` would tell you exactly that goes wrong, typically it is a conflict in libraries. As a general answer your have two options:
 - remove Homebrew  - see uninstall in their documentation

or

 - Install everything, i.e., R and package using Homebrew. The drawback is that you cannot use CRAN binaries

**Comment by Yukiwang77** [comment]
I figured it out!!! so excited :）
I had a makevar configuration in ~/.R/Makevars file set all gcc settings to 8.3.0 while my gcc has already been upgraded to 9.3. so the usr/local/Cellar/gcc directory only has 9.3.0_, not 8.3.0 any more. The Makevars file keeps telling R to compile with 8.3 which pointed to an non-exist directory. I updated the Makevars configuration and I can successfully install the package. Thanks!
